contents/revisionControl/bestPracticesGit.md

```diff
@@ -0,0 +1,259 @@
+- [torvalds/subsurfance-for-dirk](https://github.com/torvalds/subsurface-for-dirk/commits/master)
+- [torvalds/linux](https://github.com/torvalds/linux/commits/master)
+
+### Set Up Your Editor for Commit Messages
+1. To use your editor of choice for `git`-related functionality, e.g. `vim`, do one of either
+```
+git config --global core.editor "vim" # or you can do the following
+export GIT_EDITOR=vim # add to your .bashrc
+```
+2. Set your editor to wrap after 72 characters. In `vim`, you can do this by adding this to your `.vimrc`:
+```
+syntax on
+autocmd Filetype gitcommit spell textwidth=72
+```
+---
+## Always Commit Functional Code
+Merges to the following must always leave the project in a working state, i.e. it can be built and run on:
+- `master` branch, or equivalent,
+- `staging` branch, `development` branch or equivalent, if any.
+
+Changes to your own branches _that no one else is using_ can have non-functioning commits. However, you may wish to [hide the sausage making](#hide-the-sausage-making) to squash non-functioning commits into a single, functioning commit before you make a pull request.
+
+Changes to your own branches that is used by others should obey always-functioning-commits rule to minimize surprise. This is especially important if you expect your branch to be cherry-picked by another collaborator because they require a specific bit of code that you wrote.
+
+### Stashing
+If you need to switch between branches while in the middle of developing a commit, you can use the `git stash` command instead of committing your half-done code.
+
+```bash
+# stash your work not committed to HEAD yet by pushing it onto the stash stack
+git stash
+git stash push # equivalent to git stash
+
+# restore your most recently stashed work to your current working copy
+git stash pop
+
+# acts like git stash pop, but keeps a copy of the stash in the current stash stack
+git stash apply
+
+# list all stashes in the stack
+git stash list
+```
+
+Stashes are a purely local construct and cannot be pushed to a remote repository.
+
+Read more: [git stash - Saving Changes | Atlassian Git Tutorial](https://www.atlassian.com/git/tutorials/git-stash)
+
+---
+
+## One Logical Change to One Commit
+Commits are the building blocks of a codebase; each building block should contribute exactly one useful thing, like:
+
+- adding a new function or piece of data
+- fixing one or more bugs
+    - correcting incorrect behavior
+- refactoring code or data
+    - reorganizing code
+    - removing typos
+    - formatting code
+    - changing representation of data to a different format
+
+Each _logical change in code_ should translate to exactly one commit, nothing more or less. Doing so allows you to:
+- revert a particular logical change with no/less side effects
+- easily identify bad commits that caused change in behavior
+- collaborate with others, e.g. by cherry-picking a single logical change instead of finding a bit of this commit and a bit of that commit
```

Comment:
Should it be, `collaborate with others easily` instead? Since you're emphasizing the benefits, just stating `collaborate with others` doesn't seem as beneficial.